### ISSUE TITLE
Paper 9/13 — Performance–Quality Trade-off

### ISSUE THREAD

**Issue by murillo128** [murillo128/k3-out-of-core#118]
Parent: #109

**Editorial status:** `OUTLINE`

This issue coordinates the quality/semantic-cost section. It is not an execution contract.

## Purpose

Treat quality as a first-class result rather than a footnote. The paper should explicitly show what bounded cache-aware routing changes, how the effect grows over long horizons, and what the current evidence does **not** justify.

Core message:

> Bounded router-score regret constrains each local substitution, but it does not make the resulting model semantically exact. The measured systems gain therefore belongs on a performance–quality frontier, and long-horizon autoregressive feedback must be evaluated explicitly.

### 8.1 Short-horizon perturbation

Use #77 as the authority for the original controlled quality instrumentation and short-horizon mechanism:

- local MoE-output change;
- hidden-state divergence;
- logit KL/JS where relevant;
- top-token agreement;
- reference-token NLL/cross-entropy;
- teacher forcing on an exact reference token sequence.

Use this primarily to explain the measurement methodology and the fact that perturbations propagate beyond the local routed layer.

### 8.2 Long-horizon predictive drift

Primary authority: final #99.

Frozen classification:

```text
LONG_HORIZON_PREDICTIVE_DRIFT = gradual
```

Headline S2_P50 vs EXACT fixed-context result:

```text
mean reference-NLL damage      +0.012030
95% cluster-bootstrap interval  0.008440 .. 0.015435
prompt clusters                 16
```

This is a direct predictive-quality effect under the exact reference-token intervention. Do not call it task accuracy or generation quality.

Candidate figure: ΔNLL vs decode horizon, ideally showing per-family spread plus aggregate interval.

### 8.3 KNEE vs S2 quality ordering

Frozen #99 result:

```text
KNEE vs S2_P50 quality delta  +0.001090
interval                     -0.001228 .. 0.003205
classification               no_clear_difference
```

Interpretation:

> The lower-regret KNEE policy is not clearly superior to S2_P50 on the long-horizon predictive metric, despite S2's stronger physical locality result.

Avoid using this null interval as evidence of equivalence.

### 8.4 Autoregressive route feedback

Primary authority: #99 controlled direct-vs-free intervention.

Frozen result:

```text
TOKEN_MEDIATED_ROUTE_FEEDBACK = material
mean amplification            = 1.4210x
95% interval                  = 1.2225 .. 1.7099
FEEDBACK_GROWTH_TO_1024       = heterogeneous
```

Explain the intervention carefully:

- `DIRECT_FIXED_CONTEXT`: changed routing while consuming the exact reference-token trajectory;
- `FREE_TRAJECTORY`: changed routing while consuming its own generated tokens;
- the difference isolates the token-mediated feedback increment under this controlled setup.

Do not call `physical - replay` route feedback; #99 owns the causal attribution.

Candidate figure: direct vs free-trajectory effect across horizon for the three frozen bridge prompts.

### 8.5 Can simple routing statistics predict semantic damage?

Use #99's negative results as an explicit scientific result:

```text
CUMULATIVE_REGRET_PREDICTIVE      = weak
RAW_REGRET_ADDS_SIGNAL             = weak
PERTURBED_FRACTION_ADDS_SIGNAL     = no
FOLLOWUP_ROUTING_DESIGN_JUSTIFIED  = no
```

Suggested interpretation:

> Hard local regret bounds are operationally useful for constraining individual substitutions, but simple accumulation of those local quantities is not a sufficient long-horizon semantic safety proxy.

This is valuable because it prevents the paper from pretending the chosen router has a solved quality controller.

### 8.6 Task-level capability

This subsection remains `NEEDS_RESULT` until #100/#101 produce accepted evidence.

Current intended roles:

- #100: paired EXACT vs S2_P50 on 30 preregistered GPQA Diamond items, plus full 198-item S2 run against Moonshot's 93.5% only as `OFFICIAL_PROTOCOL_NEAR_MATCH`.
- #101: independent standard-suite validation, with MMLU-family likelihood scoring for direct predictive effects and GSM8K-family generation for autoregressive feedback-sensitive task behavior.

Do not write a task-quality conclusion from #99 internal metrics.

## Main-paper figure candidates

1. Long-horizon ΔNLL/drift.
2. Direct fixed-context vs free-trajectory feedback amplification.
3. Final task-level paired result from #100 and/or bounded #101 result, once available.

If page-constrained, combine 1 and 2; keep detailed per-prompt trajectories in appendix.

## Wording boundaries

Do not say:

- "quality preserved" from ΔNLL alone;
- "semantically equivalent" from a non-significant KNEE/S2 difference;
- "1.42x worse quality" — the 1.4210x value is a feedback amplification factor under the controlled metric/intervention;
- "regret does not matter" — the supported result is that the tested cumulative/raw-regret summaries have weak held-out predictive value for the measured damage;
- "GPQA matches Moonshot" unless #100's final protocol-aware result actually supports that wording.

## Target section conclusion

The section should end with a balanced statement:

> The locality gain is real and physically measurable, but it is purchased with a bounded routing approximation whose predictive effect is also measurable and can be amplified by autoregressive feedback. The current evidence supports a controllable performance–quality trade-off, not semantic equivalence.

**Comment by murillo128** [comment]
## Editorial update — Performance–Quality Trade-off drafted

Updated `paper/paper.md` on `paper/manuscript-v0` in commit `eb8b54b0ed8d1e845f73c493ae3eccd084378b57`.

### Editorial status

`EVIDENCE_CHECK` — the predictive-quality section is now drafted. Task-level capability remains explicitly pending #100/#101.

### Decisions/results reflected in the manuscript

- **Short-horizon mechanism (#77):** use teacher-forced exact reference tokens to show the propagation chain from intentional expert substitution through MoE output, hidden state, later routing/logits, and reference-token likelihood. Retained 24-token frontier traces are described as internal/predictive perturbation evidence; top-1 agreement of 23/24 is not treated as semantic or task equivalence.
- **Long-horizon predictive effect (#99):** S2_P50 vs EXACT direct fixed-context mean reference-token ΔNLL `+0.012030`, 95% cluster-bootstrap interval `0.008440..0.015435`, across 16 prompt clusters. Drift is classified `gradual`; the registered breakpoint evidence is weak rather than a supported abrupt threshold.
- **KNEE vs S2_P50:** long-horizon quality delta `+0.001090`, interval `-0.001228..0.003205`, retained as `no_clear_difference`. The text explicitly avoids equivalence wording and notes that tighter local regret does not produce a clearly ordered long-horizon predictive result.
- **Direct vs free autoregressive feedback (#99):** `DIRECT_FIXED_CONTEXT` and `FREE_TRAJECTORY` are kept distinct. Token-mediated feedback is `material`, with mean amplification `1.4210×` and interval `1.2225..1.7099`; growth to 1024 tokens is `heterogeneous`. The manuscript explicitly states that this is a controlled amplification factor, not “1.42× worse quality,” and does not derive feedback from physical-vs-replay differences.
- **Negative regret-predictor result (#99):** the section now reports the held-out hierarchy: corrected-regret-only baseline LOFO R² `0.0169`; + cumulative corrected regret `0.1513` (`weak`); + signed raw regret `0.1800` (`weak`); adding changed-slot/perturbed-layer fractions falls to `0.1423` (`no` added signal); depth conditioning reaches `0.1306` and is unsupported. Interpretation: local regret remains an operational hard bound, but simple accumulated summaries are not a sufficient long-horizon semantic-safety proxy; no follow-up routing design is justified by these results.
- **Task-level boundary:** #100 still states that no GPQA outcome has been generated/inspected; #101 remains investigation-required. The manuscript does not convert ΔNLL, top-token agreement, amplification, or predictor metrics into task accuracy and leaves Moonshot GPQA 93.5% only as an external `OFFICIAL_PROTOCOL_NEAR_MATCH` reference for #100.

### Section conclusion

The drafted section positions S2_P50 as a measured K3 performance–quality trade-off: bounded routing slack reduces physical service demand at fixed measured cache capacity, but it measurably changes the predictive trajectory and free-generation feedback can amplify that change. The supported claim is not semantic equivalence or task-quality preservation.
